impl/src/rendering/templates/partials/rust_library.template

```diff
@@ -10,10 +9,10 @@ alias(
 )
 
 {% endif -%}
-# buildifier: leave-alone
 rust_library(
     name = "{{ target_name_sanitized }}",
-    crate_type = "{{ target.kind }}",
+{% include "templates/partials/common_attrs.template" %}
+    # buildifier: leave-alone
```

Comment:
@PiotrSikora For this and the `rust_binary` template, I'm not sure how to solve this. I think it'd be better to move all sorting into the templates if possible but I don't know how realistic that is. Is this something you could take a look at?

Reply:
Could we merge and sort `crate.default_deps.dependencies` and `crate.raze_settings.additional_deps` into some temporary list before generating those targets?

Reply:
Maybe? I think it'd be nice to be able to express this within the templating but it also would make sense to have a single variable with all dependencies. Is this something you have time to try?

Reply:
I know nothing about Jinja, but does something like this work?
```
{%- set deps = [] %}
{%- for dependency in crate.default_deps.dependencies %}
{%-   deps.append(dependency.buildable_target) %}
{%- endfor %}
{%- for dependency in crate.raze_settings.additional_deps %}
{%-   deps.append(dependency) %}
{%- endfor %}
{%- for dependency in deps|sort %}
    "{{dependency}}",
{%- endfor %}
```

Reply:
I tried something similar to that. The templating tool being used here is https://github.com/Keats/tera which has a slightly different set of rules. I ran into a bunch of syntax errors trying to figure this out and got frustrated and added back `leave-alone` 😓 

In theory something like that should work and should be super simple but for whatever reason I wasn't able to get it working right away.

Reply:
@UebelAndre This works:
```
--- a/impl/src/rendering/templates/partials/rust_library.template
+++ b/impl/src/rendering/templates/partials/rust_library.template
@@ -17,10 +17,7 @@ rust_library(
         {%- if crate.build_script_target %}
         ":{{ crate_name_sanitized }}_build_script",
         {%- endif %}
-        {%- for dependency in crate.default_deps.dependencies %}
-        "{{dependency.buildable_target}}",
-        {%- endfor %}
-        {%- for dependency in crate.raze_settings.additional_deps %}
+        {%- for dependency in crate.default_deps.dependencies | map(attribute="buildable_target") | concat(with=crate.raze_settings.additional_deps) | sort %}
         "{{dependency}}",
         {%- endfor %}
     ]
```
Could you add this and remove `# buildifier: leave-alone`?

But I don't think we have any example that actually uses both.

Reply:
There's still one defect
```diff
diff --git a/examples/vendored/complicated_cargo_library/cargo/vendor/regex-0.2.11/BUILD.bazel b/examples/vendored/complicated_cargo_library/cargo/vendor/regex-0.2.11/BUILD.bazel
index ba728e02..a5c72393 100644
--- a/examples/vendored/complicated_cargo_library/cargo/vendor/regex-0.2.11/BUILD.bazel
+++ b/examples/vendored/complicated_cargo_library/cargo/vendor/regex-0.2.11/BUILD.bazel
@@ -66,12 +66,12 @@ rust_library(
     ],
     version = "0.2.11",
     deps = [
+        "//vendored/complicated_cargo_library/cargo:specs",
         "//vendored/complicated_cargo_library/cargo/vendor/aho-corasick-0.6.10:aho_corasick",
         "//vendored/complicated_cargo_library/cargo/vendor/memchr-2.3.3:memchr",
         "//vendored/complicated_cargo_library/cargo/vendor/regex-syntax-0.5.6:regex_syntax",
         "//vendored/complicated_cargo_library/cargo/vendor/thread_local-0.3.6:thread_local",
         "//vendored/complicated_cargo_library/cargo/vendor/utf8-ranges-1.0.4:utf8_ranges",
-        "//vendored/complicated_cargo_library/cargo:specs",
     ],
 )

```

Reply:
OK, I think we need to keep `# buildifier: leave-alone` for this one. `/` precedes `:` in the ASCII table, so the output is now sorted correctly, but it doesn't match `buildifier`'s sorting, which seems to be aware of the packages' hierarchy. Sadly, this means that we won't be able to solve this using templates.

Please re-add `# buildifier: leave-alone` above `deps` and this should be good to merge. Thanks!

Reply:
I did my best 😅

maybe in the future we can have the tool check to see if the buildifier binary is available and run that on the generated files.

I dropped the last two commits which is effectively reapplying `leave-alone`.